code/modules/holomap/holomap.dm

```diff
@@ -0,0 +1,357 @@
+/*
+	Somone should totally add an action button to select a location when observing the map
+	and it display a list to select from (TGUI maybe!? color coded !? YES?)
+	then show an icon like YouAreHere there
+*/
+
+///this will hold images rendered at roundstart
+GLOBAL_LIST_EMPTY(holomaps)
+
+/obj/machinery/holomap
+	name = "Holomap"
+	desc = "It displays a holographic map of the station."
+	icon = 'icons/obj/terminals.dmi'
+	icon_state = "map"
+	anchored = TRUE
+	idle_power_consumption = 5
+	active_power_consumption = 15
+	///this is so you can have holomap display other zlevels, default is the one its on
+	var/zlevel_rendered
+	//for mappers who want to add holomaps for off station locations, otherwise it defaults to the entire zlevel in Initialize()
+	//use sparingly, dont make regions different only by a few values, just make it a single one across both instances (they are very costly to make and store)
+	///should be a list {start x, start y, end x, end y} start should be bottom left corner; end should be top right... everything inside gets rendered (out of 255)
+	var/list/region_selection = list(1,1,255,255)
+	///this holds all the people it is currently displaying to
+	var/list/display_targets = list()
+	///this is set in Initialize with the correct offsets of the machine's position in the zlevel
+	var/image/YouAreHere
+	//this is the conditioned icon from the cache that will be displayed to users
+	var/image/holomap_projection
+
+/obj/machinery/holomap/Initialize(mapload)
+	. = ..()
+
+	zlevel_rendered = z
+
+	var/icon/base_icon
+	var/icon_key = "[zlevel_rendered]_[region_selection[1]]-[region_selection[2]]-[region_selection[3]]-[region_selection[4]]"
+
+	if(GLOB.holomaps[icon_key])
+		base_icon = GLOB.holomaps[icon_key]
+	else
+		base_icon = make_map()
+		var/icon/alpha_mask = icon('icons/effects/480x480.dmi', "scanline") //Scanline effect.
+		base_icon.AddAlphaMask(alpha_mask)
+		GLOB.holomaps[icon_key] = base_icon
+
+	holomap_projection = image(base_icon, loc = src, layer = ABOVE_HUD_PLANE, pixel_x = -240, pixel_y = -240)
+	holomap_projection.plane = ABOVE_HUD_PLANE
+	holomap_projection.filters = filter(type = "drop_shadow", size = 8, color = "#ffffffd4")
+	YouAreHere = image('icons/effects/holomap_icons.dmi', src, "YouAreHere", 22.1 , 2, -256+(x*2), -256+(y*2))
+	YouAreHere.plane = ABOVE_HUD_PLANE + 0.1
+	CHECK_TICK //hopefully Im using this correctly
```

Comment:
Avoid `CHECK_TICK` inside of `Initialize()`.

`Initialize()` isn't meant to sleep, and actually counts as a fail.
*I still need to finish that unit test but that can wait*